src/targets/gpu/target.cpp

```diff
@@ -104,6 +102,8 @@ std::vector<pass> target::get_passes(migraphx::context& gctx, const compile_opti
     {
         enable_pass(options.split_single_dyn_dim, split_single_dyn_dim{}),
         enable_pass(options.split_single_dyn_dim, dead_code_elimination{}),
+        enable_pass(options.split_single_dyn_dim, promote_literals{}),
```

Comment:
This pass should run before `write_literals` and it should be enabled unconditionally.